test/integration/error_output.cc

```diff
@@ -233,6 +233,29 @@ TEST(ErrorOutput, ElementErrorOutput)
   ASSERT_EQ(errors.size(), 1u);
   EXPECT_NE(std::string::npos, errors[0].Message().find(
       "Missing element description for [nonExistentElement]"));
+  errors.clear();
+  // Check nothing has been printed
+  EXPECT_TRUE(buffer.str().empty()) << buffer.str();
+
+  ASSERT_THROW(elem->RemoveChild(sdf::ElementPtr(), errors),
+               sdf::AssertionInternalError);
+  ASSERT_EQ(errors.size(), 0u);
+  buffer.str(std::string());
+
+  elem->AddAttribute("key", "std::string", "test", true, errors, "");
+  ASSERT_EQ(errors.size(), 0u);
+  elem->Get<int>(errors, "key");
+  ASSERT_EQ(errors.size(), 1u);
+  EXPECT_NE(std::string::npos, errors[0].Message().find(
+      "Invalid argument. Unable to set value [test] for key[key]."));
```

Comment:
So the reason for this seems a bit complicated: basically when the key exists but the type does not match, the call to the `Element->Get()` function ends up in the `Parm->Get()` [here](https://github.com/gazebosim/sdformat/blob/4464fbd413ff568f9a1c635f71fbaa4fae6b63e5/include/sdf/Param.hh#L729) where it tries to get the not-matching type on the `Get` call and the current value of the `param` and tries to set it with the new type that was asked for in the initial `Get` function. I don't know what's the use case for this, maybe because the type could have been wrongly parsed initially? 

IMHO if this is a wrong behavior and it is to be changed it probably deserves it's own issue, for now I'll add a comment with a clarification on the test.

Reply:
As per video call, I opened a separated issue [here](https://github.com/gazebosim/sdformat/issues/1257) so we can track if we either modify the error message of `ValueFromStringImpl` into something more generic or we somehow modify the behavior of [the Get method](https://github.com/gazebosim/sdformat/blob/4464fbd413ff568f9a1c635f71fbaa4fae6b63e5/include/sdf/Param.hh#L712).